common/changes/@microsoft/api-extractor/add-support-for-exports-types_2024-05-26-00-41.json

```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft/api-extractor",
+      "comment": "Improve support for resolving the `tsdoc-metadata.json` to include the folder referenced by a `types` field in an `\"exports\"` field and an `\"typesVersions\"` field in addition to `\"types\"`, `\"typings\"`, and `\"tsdocMetadata\"` fields.",
```

Comment:
This is technically a breaking change to the specification, since if a package built using older API Extractor is consumed by a package being built using this new API Extractor, the discovery of `tsdoc-metadata.json` could be broken, since the new API Extractor may look in a different place from where the old API Extractor wrote the file.

The actual risk of regression is perhaps relatively minor however.

- The new `<lookup>` rules will produce different results only in certain edge cases involving `exports` or `typesVersions`
- In the original intended use cases, all packages participating in this `tsdoc-metadata.json` analysis are normally compiled together by the same developer.
- @Josmithr has argued recently against processing TSDoc from another developer's NPM package at all:
   #4425 
   PR #4529 
   PR #4498

Reply:
Since this PR adds new lookup locations, but does not remove any old lookup locations, it would only break compatibility if the library was built with a new API Extractor, but its consumer was built with an old API Extractor.